teps/0093-add-sign-verify-subcommand-to-the-cli.md

```diff
@@ -0,0 +1,149 @@
+---
+status: proposed
+title: Add sign and verify subcommand to the CLI
+creation-date: '2022-10-05'
+last-updated: '2022-10-05'
+authors:
+- '@Yongxuanzhang'
+- '@nadgowdas'
+- '@lukehinds'
+collaborators: []
+---
+
+# TEP-0093: Add sign and verify subcommand to the CLI
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+- [Design Details](#design-details)
+- [Alternatives](#alternatives)
+- [Implementation Plan](#implementation-plan)
+  - [Test Plan](#test-plan)
+  - [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This proposal aims at adding `sign` and `verify` to Tekton CLI to help Trusted Resources users to sign and verify Tekton Resources.
+
+
+## Motivation
+
+There is a growing open-source ecosystem around tekton [catalog](https://github.com/tektoncd/catalog) and [artifactHUB](https://artifacthub.io), where number of tasks/pipelines are made available ready-to-use in our CI/CD. These tasks provides commonly used functions like `git-clone`, `curl`, etc. And also includes vendor specific functions like `aws-cli`, `openshift-client`. Such a marketplace is essential in practicing DRY principle. But, as we are building such an open-distribution framework, we need to ensure task distribution is secure.
+
+Objective in this proposal is to faciliate an easy and handy approach for task/pipeline providers to sign their files and for consumers to verify them using native `tkn cli` commands.
+
+Also, this complements [TEP-0091](https://github.com/tektoncd/community/blob/main/teps/0091-trusted-resources.md). TEP-0091 proposes to apply verification at the controller and the signing is suggested to be implemented into Tekton CLI. This proposal suggests an CLI extension for developers to sign and verify tasks/pipelines YAML files.
+
+### Goals
+
+1. Add the ability to run `tkn trustedresources sign` and `tkn trustedresources verify` fom the CLI. `Sign` will take the private key and the resource yaml file, sign the resource object and attach signature to the annotation.
```

Comment:
I think we can discuss the "subcommand" / command "shape/path" 👼🏼

Reply:
Sorry I don't get this comment. Could you elaborate more on it? 
What would be the suggested commands?

Reply:
@Yongxuanzhang I am not sure yet, hence me approving. I wonder if it should all be under one subcommand (`trustedresources`, `trust` or something) or if it should be something like `tkn task sign …`, `tkn {object} verify`, etc.. But in any case I am +:100: on having this on the cli 👼🏼

Reply:
Just wondering, do we need subcommand for every resource {`task`, `pipeline` ..} ? If the behavior is going to be same regardless of object type, it can be simplied with just `tkn sign/verify <resource>`

Reply:
> @Yongxuanzhang I am not sure yet, hence me approving. I wonder if it should all be under one subcommand (`trustedresources`, `trust` or something) or if it should be something like `tkn task sign …`, `tkn {object} verify`, etc.. But in any case I am +💯 on having this on the cli 👼🏼

Oh I see, I like the idea of using "tkn task sign", so users don't have to specify the resource type

Reply:
> Just wondering, do we need subcommand for every resource {`task`, `pipeline` ..} ? If the behavior is going to be same regardless of object type, it can be simplied with just `tkn sign/verify <resource>`

That would be similar to the current proposal without "trustedresources"? But still need users to specify object type